File: Baekjoon/백트래킹/15649.py
# https://www.acmicpc.net/problem/15649

import sys

input = sys.stdin.readline

N, M = map(int, input().split())
visited = [False] * N
lst = []

def DFS(depth):
    if depth == M:
        print(' '.join(map(str, lst)))
        return
    for i in range(N):
        if not visited[i]:
            visited[i] = True
            lst.append(i+1)
            DFS(depth+1)
            visited[i] = False
            lst.pop()
DFS(0)

File: Baekjoon/백트래킹/15649.cpp
// https://www.acmicpc.net/problem/15649

#include <iostream>
#include <vector>

using namespace std;

int N, M;
int arr[8];
bool visited[9];

void dfs(int cnt) {
    if (cnt == M) {
        for (int i = 0; i < M; i++) {
            cout << arr[i] << ' ';
        }
        cout << '\n';
        return;
    }

    for (int i = 1; i <= N; i++) {
        if (!visited[i]) {
            visited[i] = true;
            arr[cnt] = i;
            dfs(cnt + 1);
            visited[i] = false;
        }
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    cin >> N >> M;

    dfs(0);

    return 0;
}